utils/inpsSoapClient.ts

```diff
@@ -0,0 +1,109 @@
+import { format } from "date-fns";
+import { fromNullable } from "fp-ts/lib/Either";
+import { TaskEither, tryCatch } from "fp-ts/lib/TaskEither";
+import { readableReport } from "italia-ts-commons/lib/reporters";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import fetch from "node-fetch";
+import { DOMParser } from "xmldom";
+import { ConsultazioneSogliaIndicatoreInput } from "../generated/definitions/ConsultazioneSogliaIndicatoreInput";
+import { ConsultazioneSogliaIndicatoreResponse } from "../generated/definitions/ConsultazioneSogliaIndicatoreResponse";
+
+const SOAP_REQUEST_TEMPLATE = `<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:con="http://inps.it/ConsultazioneISEE">
+<soapenv:Header/>
+<soapenv:Body>
+  <con:ConsultazioneSogliaIndicatoreRequest>
+    <con:Richiesta DataValidita="{dataValidita}" CodiceFiscale="{fiscalCode}" FornituraNucleo="{fornituraNucleo}" CodiceSoglia="{codiceSoglia}"/>
+  </con:ConsultazioneSogliaIndicatoreRequest>
+</soapenv:Body>
+</soapenv:Envelope>`;
+
+const INPS_NAMESPACE = "http://inps.it/ConsultazioneISEE";
+
+export interface ISoapClientAsync {
+  ConsultazioneSogliaIndicatore: (
+    params: ConsultazioneSogliaIndicatoreInput
+  ) => TaskEither<Error, ConsultazioneSogliaIndicatoreResponse>;
+}
+
+export function createClient(
+  INPS_SERVICE_HOST: NonEmptyString,
+  endpoint: NonEmptyString
+): ISoapClientAsync {
+  return {
+    ConsultazioneSogliaIndicatore: (
+      params: ConsultazioneSogliaIndicatoreInput
+    ) => {
+      return tryCatch(
+        async () => {
+          const requestPayload = SOAP_REQUEST_TEMPLATE.replace(
```

Comment:
I'd rather suggesto to use string literals variables and a method instead of replace:

```ts
const getSoapRequestTemplate(codiceFiscale: string, ...) => `... <>$[codiceFiscale} <> ...`
```

Reply:
fixed in 0c41302